### PR TITLE
fix: update seedlot existing pages

### PR DESCRIPTION
**Pull request by craigyu**
# Description
Jira ticket: [FSADT2-547](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-547)

This PR consists mostly visual updates and some components changes.
Shout-out to @mgaseta for his help!

### Notable Changes

- All "Exisiting seedlot" files and class names have been updated to "My seedlots" for consistency
- the `EmptySection` component can now accept either a `Icon` or `Pictogram`. If both props are provided, `Pictogram` will be used
- `frontend/src/utils/filterUtils.ts` has been added, it's useful for Carbon input filtering
- the `ComboButton` component has an optional `menuOptionsClass` prop for style fine tuning

### How to test

1. Go to http://localhost:3000/seedlot
    - The title for the third card and the table should be "My seedlots" instead of "Existing seedlot" 
image

2. Go to `frontend/src/components/SeedlotActivities/index.tsx`
    - on line 118, replace `cards.map` with `empty_cards.map`
    - on the same web page, the "My seedlot" and "Activity history" cards should look like this: 
image
3. On the same web page, click on the "Register an A class seedlot" card
    - the "Applicant agency name" and "Applicant agency number"'s helper text are updated
    - the "Applicant agency number" input is now a `ComboBox`, you can test the filter function by typing
        - note that the `ComboBox` will have a 'x' sign to clear the current field, which is inconsistent with the design, hopefully that's ok with the design team
4. Fill in the necessary fields and click on the "Create seedlot number" button
    - you should be redirected to the success page, and the success message should have the seedlot number we just generated 
image
5. Click on the "Go to seedlot's detail screen" button
    - The blue combo button on the right should look like this: 
image
6. Go to `frontend/src/components/FormReview/index.tsx`
    - on line 65, replace `mockFormData.length` with `emptyMockFormData.length`
    - on the same web page, the "Form Review" section should look like this: 
image

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
Tested locally and ran `yarn run lint`

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-webapp-70-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-webapp/actions/workflows/merge-main.yml)

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Frontend](https://nr-spar-webapp-70-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-spar-webapp/actions/workflows/merge-main.yml)

### CODE REVIEW

**Comment by craigyu** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=bcgov_nr-spar-webapp&pullRequest=70)
> 
> [Bug](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=CODE_SMELL) [4 Code Smells](https://sonarcloud.io/project/issues?id=bcgov_nr-spar-webapp&pullRequest=70&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=bcgov_nr-spar-webapp&pullRequest=70&metric=new_coverage&view=list) [58.2% Coverage](https://sonarcloud.io/component_measures?id=bcgov_nr-spar-webapp&pullRequest=70&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=bcgov_nr-spar-webapp&pullRequest=70&metric=new_duplicated_lines_density&view=list) [16.3% Duplication](https://sonarcloud.io/component_measures?id=bcgov_nr-spar-webapp&pullRequest=70&metric=new_duplicated_lines_density&view=list)

The duplication is due to the test mock data 💀, they can be removed once the PR is approved

**Review by RMCampos — APPROVED** [review]
Great job 🥳

**Comment by craigyu** [comment]
@mgaseta fixed lints!

**Review by lucasbuenoencora — APPROVED** [review]
Great Job! :rocket:

File: frontend/src/components/ApplicantInformation/index.tsx [hunk 1]
```diff
@@ -23,10 +24,21 @@ import GeneticClassesType from '../../types/GeneticClasses';
 import getUrl from '../../utils/ApiUtils';
 import ApiAddresses from '../../utils/ApiAddresses';
 import { useAuth } from '../../contexts/AuthContext';
+import { FilterObj, filterInput } from '../../utils/filterUtils';
 
 import './styles.scss';
 
+interface ComboBoxEvent {
+  selectedItem: any
```
Comment (mgaseta) [inline frontend/src/components/ApplicantInformation/index.tsx:32]: Maybe trying using a different type here instead of `any`, I guess it can be `string|null`.

File: frontend/src/components/Card/StandardCard/index.tsx [hunk 2]
```diff
@@ -33,19 +39,31 @@ const StandardCard = ({
...
+                <Image className="std-card-pictogram" />
+              ) : (
+                <ActivityHistory
+                  history={ActivityHistoryItems}
+                />
+              )}
+            </div>
+          )
+      }
+
```
Comment (mgaseta) [inline frontend/src/components/Card/StandardCard/index.tsx:66]: Remove empty line.

File: frontend/src/components/EmptySection/index.tsx [hunk 3]
```diff
@@ -1,28 +1,39 @@
...
-  icon, title, description
+  icon, title, description, pictogram
 }: EmptySectionProps) => {
-  const Icon = Icons[icon];
+  let Img;
+
+  if (icon) {
+    Img = Icons[icon];
+  }
+  // if both icon and pictogram are passed in then pictogram will be used
```
Comment (mgaseta) [inline frontend/src/components/EmptySection/index.tsx:25]: Start the comment with a capital letter.

File: frontend/src/components/FormReview/index.tsx [hunk 4]
```diff
@@ -19,56 +58,31 @@ const FormReview = () => (
...
+                  key={data.id}
+                  title={(
+                    <TitleAccordion
+                      title={data.title}
+                      description={data.description}
+                    />
+                  )}
+                />
+              ))
+            }
```
Comment (mgaseta) [inline frontend/src/components/FormReview/index.tsx:77]: Indentation is a bit odd here, the opening bracket on line 65 is not aligned with the closing one on line 77, an additional tab from line 66 to line 77 should fix it.

File: frontend/src/components/FormReview/styles.scss [hunk 5]
```diff
@@ -28,3 +28,7 @@
   color: var(--#{vars.$bcgov-prefix}-text-secondary);
   padding-top: 0.25rem;
 }
+
+.form-review-empty-section .empty-section-container{
+  padding-bottom: 6rem;
+}
```
Comment (mgaseta) [inline frontend/src/components/FormReview/styles.scss:34]: Add an empty line to the end of the file.

File: frontend/src/views/Seedlot/SeedlotDetails/styles.scss [hunk 6]
```diff
@@ -29,3 +29,11 @@
...
   padding: 1.75rem 2.5rem !important;
 }
+
+.edit-seedlot-form {
+  width: 11.75rem;
+}
+
+.edit-seedlot-form .#{vars.$bcgov-prefix}--overflow-menu-options__btn {
+  max-width: 100%;
+}
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/SeedlotDetails/styles.scss:39]: Add an empty line at the end of the file.

File: frontend/src/views/Seedlot/SeedlotCreatedFeedback/index.tsx [hunk 7]
```diff
@@ -11,7 +11,7 @@ import './styles.scss';
 
 const SeedlotCreatedFeedback = () => {
   const navigate = useNavigate();
-  const { seedlot } = useParams();
+  const seedlotNumber = useParams().seedlot;
```
Comment (mgaseta) [inline frontend/src/views/Seedlot/SeedlotCreatedFeedback/index.tsx:14]: Why the change here?

**Review by mgaseta — CHANGES_REQUESTED** [review]
LGTM 🥇✔! Added only a few markups, mostly related to code style.

File: frontend/src/components/ApplicantInformation/index.tsx (hunk 1 above)
Comment (craigyu) [reply frontend/src/components/ApplicantInformation/index.tsx:32]: having null doesn't work, changed to `string`

Comment (mgaseta) [reply frontend/src/components/ApplicantInformation/index.tsx]: Great, thanks!

File: frontend/src/views/Seedlot/SeedlotCreatedFeedback/index.tsx (hunk 7 above)
Comment (craigyu) [reply frontend/src/views/Seedlot/SeedlotCreatedFeedback/index.tsx:14]: felt like calling it `seedlot` is not descriptive enough and it looks weird to me in the code 👀. 
For example:
```navigate(`/seedlot/details/${seedlot`)}```
    vs
```navigate(`/seedlot/details/${seedlotNumber}`)}```

 `Your A class seedlot {seedlot}`
    vs
 `Your A class seedlot {seedlotNumber}`

I can change it back if you prefer it the other way

Comment (mgaseta) [reply frontend/src/views/Seedlot/SeedlotCreatedFeedback/index.tsx:14]: Oh, I see, that does makes the it more descriptive. Let's leave like this then.

Comment (craigyu) [reply frontend/src/views/Seedlot/SeedlotCreatedFeedback/index.tsx:14]: alright my latest commit should have fixed all the issues above

**Review by mgaseta — APPROVED** [review]
Nice job 🥇✔!
